app/views/layouts/application.html.erb

```diff
@@ -1,15 +1,13 @@
 <!DOCTYPE html>
 <html>
-  <head>
-    <title>SampleApp</title>
-    <%= csrf_meta_tags %>
-    <%= csp_meta_tag %>
-
-    <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track': 'reload' %>
-    <%= javascript_include_tag 'application', 'data-turbolinks-track': 'reload' %>
-  </head>
-
-  <body>
-    <%= yield %>
-  </body>
+<head>
+  <title><%=  yield(:title) %> | <%= t "layouts.application.page_title" %></title>
+  <%= csrf_meta_tags %>
+  <%= csp_meta_tag %>
```

Comment:
`csp_meta_tag `  cách thức hoạt động của nó như nào nhỉ

Reply:
CSP là Content Security Policy . CSP thường dùng để chống tấn công Cross-Site-Script (XSS) mà thường hay gặp khi hacker tấn công

Reply:
@tuyetnguyen123  vấn đề là nó chống như nào cơ ?

Reply:
Nghĩa là gửi và chèn lệnh và script độc hại, những mã độc này thường được viết với ngôn ngữ lập trình phía client.:((

Reply:
@tuyetnguyen123  vấn đề câu hỏi là: `csp_meta_tag`  chống lại XSS như nào ? chứ không phải đang hỏi tác dụng

Reply:
Content Security Policy (CSP) là chính sách bảo mật nội dung, được sử dụng để xác định các nguồn nội dung an toàn trên website mà trình duyệt có thể tải về cho người dùng. CSP là biện pháp đối phó rất hiệu quả với kiểu hack chèn mã độc Cross Site Scripting (XSS).
Khi bạn truy cập vào một trang web, trình duyệt sẽ gửi yêu cầu tải nội dung đến máy chủ. Máy chủ sẽ gửi trả lại nội dung của trang web này cho trình duyệt, trong đó bao gồm các file CSS, Javascript, Font, Frame … Trình duyệt sẽ tải toàn bộ những file này, vì nó được chỉ định phải làm như vậy từ mã nguồn của trang web để có thể hiển thị nội dung.
Và Hacker có thể lợi dụng điều này để đặt một đoạn mã ở trong mã nguồn ( những người sử dụng mã nguồn không rõ nguồn gốc hoặc hàng nulled rất hay bị dính ) hoặc trong một phần bình luận trên trang web để tải một số file Javascript độc hại từ một nguồn bên ngoài.
Trình duyệt sẽ không tự biết được có nên tải những file này hay không, nó chỉ thực thi theo yêu cầu, cho dù nguồn gốc của file Javascript đến từ trang hehetuilahacker.com

Đây chính là . lý do chúng ta cần đến CSP.

Tạo một danh sách CSP cho phép bạn chỉ định các nguồn nội dung an toàn mà trình duyệt có thể tải. Bằng cách này bạn có thể bảo vệ người truy cập và chính bạn khỏi các mã độc hại do Hacker cài cắm vào trang .